src/lib.rs

```diff
@@ -334,7 +423,7 @@ impl Connection<Authenticated> {
             thread::yield_now();
             let mut cur_len = wbuf.lock().unwrap().writer.buffer().len();
             if cur_len != start_len {
-                for d in wait.iter() {
+                for d in &wait {
```

Comment:
The idiom is to avoid `.iter()` methods on types that have implemented `IntoIterator` for `&T`. for-loops look for an implementation of the `Iterator` trait, falling back to `IntoIterator` if `Iterator` is not implemented